spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/TestBaseWithCatalog.java

```diff
@@ -59,18 +104,28 @@ protected static Object[][] parameters() {
   }
 
   @BeforeAll
-  public static void createWarehouse() throws IOException {
-    TestBaseWithCatalog.warehouse = File.createTempFile("warehouse", null);
-    assertThat(warehouse.delete()).isTrue();
+  public static void setUpAll() {
+    initRESTCatalog();
   }
 
   @AfterAll
-  public static void dropWarehouse() throws IOException {
+  public static void tearDownAll() throws Exception {
     if (warehouse != null && warehouse.exists()) {
       Path warehousePath = new Path(warehouse.getAbsolutePath());
       FileSystem fs = warehousePath.getFileSystem(hiveConf);
       assertThat(fs.delete(warehousePath, true)).as("Failed to delete " + warehousePath).isTrue();
     }
+    stopRESTCatalog();
+  }
+
+  private static void initRESTCatalog() {
+    restCatalog = RCKUtils.initCatalogClient(CONFIG);
```

Comment:
maybe just add
```
public Map<String, String> config() {
    return config;
  }
```
to the extension class, so that this can be changed to 
```suggestion
    restCatalog = RCKUtils.initCatalogClient(restServerExtension.config());
```